multicast/recv.py

```diff
@@ -272,6 +272,71 @@ def parseArgs(arguments=None):
 	return parser.parse_args(arguments)
 
 
+def joinstep(groups, port, iface=None, bind_group=None, isock=None):
+	"""Will join the given multicast group(s).
+
+	The JOIN function. Will start to listen on the given port of an interface for multicast
+	messages to the given group(s).
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+		>>> import multicast
+		>>> multicast.recv is not None
+		True
+		>>>
+
+	Testcase 1: Stability testing.
+
+		>>> import multicast
+		>>>
+		>>> multicast.recv is None
+		False
+		>>>
+		>>> multicast.recv.joinstep is None
+		False
+		>>> type(multicast.recv.joinstep)
+		<class 'function'>
+		>>> multicast.recv.joinstep(None, 59991) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<socket.socket...>
+		>>> tst_fxtr = multicast.__MCAST_DEFAULT_GROUP
+		>>> multicast.recv.joinstep([tst_fxtr], 59991) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<socket.socket...>
+		>>> multicast.recv.joinstep(
+		... 		[tst_fxtr], 59991, None, tst_fxtr
+		... ) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<socket.socket...>
+		>>> sk_fxtr = genSocket()
+		>>> multicast.recv.joinstep(
+		... 		[tst_fxtr], 59991, None, tst_fxtr, sk_fxtr
+		... ) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<socket.socket...>
+		>>> sk_fxtr.close()
+		>>>
+
+
+	"""
+	if groups is None:
+		groups = []
+	if isock is None:
+		sock = genSocket()
+	else:
+		sock = isock.dup()
+	try:
+		sock.bind(('224.0.0.1' if bind_group is None else bind_group, port))
```

Comment:
🤔 This needs to be documented and should be refactored to be easier to maintain in the future.
- [ ] track lack of documentation issue.
- [ ] track need to refactor hard-coded RFC compliant default.

Reply:
Documentation should be part of issue #25

Reply:
🤔 Not sure this does need refactor, this is a failsafe default, just document it for now.